### PR TITLE
fix(sprint-poker): Select All issues

### PR DESCRIPTION
**Pull request by nickoferrall**
Fix https://github.com/ParabolInc/parabol/issues/6213

### CODE REVIEW

File: packages/server/graphql/mutations/updatePokerScope.ts
Comment (nickoferrall) [inline packages/server/graphql/mutations/updatePokerScope.ts:169]: We were looping through all the updates in `additiveUpdatesWithTaskIds` and then using `dimensions` to get the `newStages`. However, we would then overwrite `newStageIds` instead of appending them. 

This meant that only one of two (depending on the number of dimensions) `newStageIds` would be returned from the mutation.

File: packages/client/mutations/UpdatePokerScopeMutation.ts
Comment (nickoferrall) [inline packages/client/mutations/UpdatePokerScopeMutation.ts:124]: When selecting all issues, `stages` will include many of the same issues as `newStages` so it will duplicate the issues. 

We could make sure that `nextStages` only contains unique stages, but it looks like the `updater` here isn't very helpful so I'd prefer to remove it. I've tested updating the poker scope with and without the updater with two users by selecting & deselecting all as well as selecting & deselecting individual issues. The behaviour is identical either way.

File: packages/client/types/constEnums.ts
Comment (nickoferrall) [inline packages/client/types/constEnums.ts:369]: This results in ugly logic elsewhere, but I get an error if I select/deselect 50 GitLab issues: `Query too large`. This doesn't happen with the GitHub or Jira integration. The reason is the [GitLab query](https://github.com/ParabolInc/parabol/blob/5235acc69ab0094009a15832ad9f5344a11024b1/packages/client/components/GitLabScopingSearchResults.tsx) is more complex as it needs to grab projects and then issues within those projects rather than just querying the issue. 

The complex query will be fixed in this issue: https://github.com/ParabolInc/parabol/issues/6325 I've updated the AC to include removing this.

File: packages/client/hooks/useUnusedRecords.ts
Comment (BartoszJarocki) [inline packages/client/hooks/useUnusedRecords.ts:14]: +1 you could make the `tresh` param mandatory and provide `Threshold.MAX_POKER_STORIES` as default, then this line could be removed

File: packages/client/mutations/UpdatePokerScopeMutation.ts
Comment (BartoszJarocki) [reply packages/client/mutations/UpdatePokerScopeMutation.ts:125]: i tested and it works. out of curiosity, do we where the relay store is updated so it actually shows selected/deselected issues without this updater? i suspect it'd be a meeting subscription. i just want to make sure we don't rely on some accidental side effect

File: packages/client/types/constEnums.ts
Comment (BartoszJarocki) [reply packages/client/types/constEnums.ts:372]: +2 i appreciate the context!

**Review by BartoszJarocki — APPROVED** [review]
lgtm!

File: packages/client/mutations/UpdatePokerScopeMutation.ts
Comment (nickoferrall) [reply packages/client/mutations/UpdatePokerScopeMutation.ts:125]: I'm not sure. It might be because the local data update isn't very complex, but I'd like to get the Maintainer's input on this

File: packages/server/graphql/types/EstimateStage.ts
Comment (nickoferrall) [inline packages/server/graphql/types/EstimateStage.ts:136]: This fixes the error mentioned in https://github.com/ParabolInc/parabol/issues/6511 where a user can't join a meeting. 

If there are many issues selected, GitLab will stop returning `issueData` and return `issueError` saying that the query complexity has exceeded the limit of 250. By returning a null field here, we prevent the app from crashing.

However, if you select all issues in GitLab and refresh the page, GitLab can't retrieve all the issues and shows the dummy error text in the sidebar. I'm working on fixing this in https://github.com/ParabolInc/parabol/pull/6520

File: packages/server/graphql/types/EstimateStage.ts
Comment (Dschoordsch) [inline packages/server/graphql/types/EstimateStage.ts:137]: +1 please remove

File: packages/client/mutations/UpdatePokerScopeMutation.ts
Comment (Dschoordsch) [reply packages/client/mutations/UpdatePokerScopeMutation.ts:125]: Would love to look into the actual implementation, but basically because the objects in "stages" and "newStages" have the same id they are treated as identical, i.e. merged in the store.

**Review by Dschoordsch — APPROVED** [review]
Tested, works for the most part, one small issue when we selected already 50 issues with a different integration and then switch to the GitLab tab it has -35 issues remaining:
Screenshot from 2022-05-11 12-42-22
Since this is an edge case and it works better than before, I'm fine with merging as is and fixing in a follow up

**Comment by nickoferrall** [comment]
Thanks for the review! 

> when we selected already 50 issues with a different integration and then switch to the GitLab tab it has -35 issues remaining

This issue will be fixed in this PR: https://github.com/ParabolInc/parabol/pull/6541

**Pull request merged**
